components/brave_news/browser/feed_v2_builder.cc

```diff
@@ -310,11 +313,94 @@ std::vector<mojom::FeedItemV2Ptr> GenerateChannelBlock(
   }
 
   std::vector<mojom::FeedItemV2Ptr> result;
-  result.push_back(mojom::FeedItemV2::NewCluster(
-      mojom::Cluster::New("channel", channel, std::move(article_elements))));
+  result.push_back(mojom::FeedItemV2::NewCluster(mojom::Cluster::New(
+      mojom::ClusterType::CHANNEL, channel, std::move(article_elements))));
   return result;
 }
 
+// Generate a Topic Cluster block
+// https://docs.google.com/document/d/1bSVHunwmcHwyQTpa3ab4KRbGbgNQ3ym_GHvONnrBypg/edit#heading=h.4vwmn4vmf2tq
+std::vector<mojom::FeedItemV2Ptr> GenerateTopicBlock(
+    const Publishers& publishers,
+    base::span<Topic>& topics) {
+  if (topics.empty()) {
+    return {};
+  }
+  DVLOG(1) << __FUNCTION__;
+  auto result = mojom::Cluster::New();
+  result->type = mojom::ClusterType::TOPIC;
+
+  auto [topic, articles] = std::move(topics[0]);
```

Comment:
A `span` is merely a reference, so this is actually modifying the original `topics_` item, and leaving a moved from object in it, which is not what I assume you would like to do.

Please, take a reference as:

```c++
const auto& topic = topics.front();
```

Reply:
What I'm trying to say is that you want to avoid any `move` operations, I think, to not change the source data. Again, passing the argument as `base::span<const Topic>` would also guarantee that that doesn't occur by accident.

Reply:
:facepalm: I actually knew this - I guess I wasn't thinking at all when I changed it to a `span`. Great catch @cdesouza-chromium (and thanks for all your help!).